### DISCUSSION TITLE
How do you test structured data / schema markup from localhost?

### DISCUSSION THREAD

**Discussion by dungle-scrubs** [vercel/next.js#40972] [Help] [answered]
I'm attempting to validate structured data (using Google's Rich Results Test and Schema.org's Schema Markup Validator) from a local development server.

My only idea is to use ngrok to create a web-accessible live link for Google to crawl.  However, the page accessed through ngrok is modifying the JSON inside of each `<script type="application/ld+json">` such that double quotes are replaced with html entities.

This throws parsing errors inside the validators, so I currently can't text any schema without running a full build. 

How do you approach this? Cheers!

**Comment by icyJoseph** [comment, marked as answer]
Maybe you could try with https://theboroer.github.io/localtunnel-www/ , https://github.com/localtunnel/localtunnel, hopefully it doesn't cause sequences to be escaped.

**Reply by dungle-scrubs** [reply]
Thanks, the problem doesn't seem to exist with localtunnel.  Appreciate the help!

**Comment by Adil512** [comment]
Our company Brand eCommerce website is built completely on Next.js and for schema, we are using online schema generator tools. And it is working perfectly. I created all the required schemas and provided them to our developer, who applied them manually. Can use any online Schema Generator tool like **[https://schemabuilder.net/](https://schemabuilder.net/)**
